modules/phase_field/src/userobjects/PolycrystalHex.C

```diff
@@ -33,21 +33,41 @@ PolycrystalHex::PolycrystalHex(const InputParameters & parameters)
     _x_offset(getParam<Real>("x_offset")),
     _perturbation_percent(getParam<Real>("perturbation_percent"))
 {
+  if (_columnar_3D == false && _dim == 3)
+    mooseError(
+        "PolycrystalHex is supported on 2D domains or 3D domains with the columnar_3D option");
+  if (_grain_num % 2 != 0)
+    mooseError("PolycrystalHex requires an even square number for 2D or columnar 3D");
+
   _random.seed(_tid, getParam<unsigned int>("rand_seed"));
 }
 
 void
 PolycrystalHex::precomputeGrainStructure()
 {
-  const unsigned int root = MathUtils::round(std::pow(_grain_num, 1.0 / _dim));
+  unsigned int d = _dim;
+
+  if (_columnar_3D && _dim == 3)
+    d -= 1;
+
+  // check if periodic boundary condition is set
+  for (unsigned int j = 0; j < d; ++j)
+  {
+    for (unsigned int i = 0; i < _vars.size(); ++i)
+      if (!_mesh.isTranslatedPeriodic(_vars[i]->number(), j))
+        mooseError("PolycrystalHex works only with periodic BCs");
```

Comment:
Why are we making it mandatory to use PBC?

Reply:
Phase-field models tend to use hexagonal grains to have a structure that's stable against grain growth. Without PBC, some of the grains generated are not six sided. This is likely to result in grain growth and may not serve the main purpose. Making it mandatory to use PBC can prevent users from inadvertently creating undesired microstructures

Reply:
@SudiptaBiswas are we good to go here?